fgpyo/util/inspect.py

```diff
@@ -9,24 +9,85 @@
 
 if sys.version_info >= (3, 8):
     from typing import Literal
+    from typing import Protocol
 else:
     from typing_extensions import Literal
+    from typing_extensions import Protocol
 if sys.version_info >= (3, 12):
     from typing import TypeAlias
 else:
     from typing_extensions import TypeAlias
 
+import dataclasses
 import functools
+from dataclasses import MISSING as DATACLASSES_MISSING
+from dataclasses import fields as get_dataclasses_fields
+from dataclasses import is_dataclass as is_dataclasses_class
 from enum import Enum
 from functools import partial
 from pathlib import PurePath
+from typing import TYPE_CHECKING
 from typing import Callable
 from typing import Optional
-
-import attr
+from typing import TypeVar
 
 import fgpyo.util.types as types
 
+try:
+    import attr
+
+    _use_attr = True
+    from attr import NOTHING as ATTR_NOTHING
+    from attr import fields as get_attr_fields
+    from attr import fields_dict as get_attr_fields_dict
+
+    Attribute = attr.Attribute
+
+    MISSING = {DATACLASSES_MISSING, ATTR_NOTHING}
+except ImportError:
+    _use_attr = False
+    attr = None
```

Comment:
Could you document what this variable is for? It is in the public namespace and I don't see an equivalent of it in the `import attr` success branch.

Reply:
I added documentation. This is the success branch, and there is actually a corresponding value on the import failure branch.